pkg/server/proto/server.proto

```diff
@@ -4489,6 +4489,10 @@ message Token {
     Trigger trigger = 10;
   }
 
+  // A client that sees this populated will use the details to fetch a token
+  // via oauth instead of submitting this token directly.
+  OAuthCredentials oauth_creds = 12;
```

Comment:
I'm trying to think if this should be in an `ExternalAuthenticator` oneof to hedge against future non-oauth auth options, but I think this is fine.

Reply:
Good call, did this!